extensions-contrib/opentelemetry-extensions/src/main/java/org/apache/druid/data/input/opentelemetry/protobuf/OpenTelemetryMetricsProtobufReader.java

```diff
@@ -145,14 +145,16 @@ private List<InputRow> parseMetric(Metric metric, Map<String, Object> resourceAt
       case SUM: {
         inputRows = new ArrayList<>(metric.getSum().getDataPointsCount());
         metric.getSum()
-            .getDataPointsList()
+            // https://github.com/open-telemetry/opentelemetry-collector/blob/2ebe1e501f9a206237de1009a127f1a419ab3696/pdata/internal/data/protogen/metrics/v1/metrics.pb.go#L123-L139
+            .getDataPointsList().stream().filter(x -> (x.getFlags() & 1) != 1)
             .forEach(dataPoint -> inputRows.add(parseNumberDataPoint(dataPoint, resourceAttributes, metricName)));
```

Comment:
it might be simpler to add the code to parseNumberDatapoint to avoid repeating ourselves for every metric type.

Reply:
I was thinking that as well, but I wasn't sure what we would do once we've detected the set flag at that level. Would we return `null`, set the value as `null` or how do you suggest handling it there?

Reply:
What about having a method
```java
static boolean hasRecordedValue(Datapoint d) {
  return d.getFlags() & DATA_POINT_FLAGS_NO_RECORDED_VALUE == 0
}
```